browser/src/control/jsdialog/Widget.DrawingArea.js

```diff
@@ -243,12 +245,25 @@ function _drawingAreaControl (parentContainer, data, builder) {
 				keyCode |= UNOModifier.CTRL;
 			}
 
-			builder.callback('drawingarea', 'keypress', container, keyCode, builder);
+			callback('drawingarea', 'keypress', container, keyCode, builder);
 		}
 
 		event.preventDefault();
 	});
 
+	// Fix for LibreOfficeKit crash: "sendDialogEvent: no widget id set"
+	// Problem: During rapid mouse interactions on drawing areas (Writer->Format->Page Style->Area->Pattern->Pattern Editor),
+	// the JSDialog system's _updateWidgetImpl function can clear container.id (set to '') during
+	// widget updates, creating a race condition. When event handlers then call builder.callback()
+	// with the container object, it has an empty ID, causing dialog events to be sent with
+	// empty widget IDs to the core, which triggers an assertion failure.
+	// This wrapper ensures container.id is restored from the original data.id before
+	// any callback is made, preventing empty widget IDs from being sent in dialog events.
+	function callback(...args) {
```

Comment:
This is rather a workaround than a fix -> can we get example of such behavior? how to reproduce?
I belive we do something wrong and we replace DOM nodes incorrectly during updates.